mlir/examples/standalone/test/python/smoketest_really_alone.py

```diff
@@ -0,0 +1,37 @@
+# RUN: echo "do nothing"
+# just so lit doesn't complain about a missing RUN line
+# noinspection PyUnusedImports
+import contextlib
+import ctypes
+import sys
+
+
+@contextlib.contextmanager
+def dl_open_guard():
+    old_flags = sys.getdlopenflags()
+    sys.setdlopenflags(old_flags | ctypes.RTLD_GLOBAL)
```

Comment:
we need to use `RTLD_GLOBAL` so that `libMLIRPythonCAPI`'s symbols are made available for relocation processing of any other object (i.e., `_standReallyAloneDialectsNanobind`).